assembly/src/parsers/field_ops.rs

```diff
@@ -169,20 +169,52 @@ pub(super) fn parse_pow2(
 /// Appends a sequence of operations to raise the base(second element) to the power specified by
 /// the top element in the stack.
 ///
-/// VM cycles - 73 cycles
+/// VM cycles per mode
+/// - exp: 73 cycles
+/// - exp.uxx: 9 + xx cycles
+/// - exp.b: 10 + Ceil(log2(b))
 pub(super) fn parse_exp(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
-    if op.num_parts() > 1 {
-        return Err(AssemblyError::extra_param(op));
+    #[allow(unused_assignments)]
+    let mut n = 64;
+
+    match op.num_parts() {
+        1 => {
+            n = 64;
+        }
+        2 => {
+            let param_value = op.parts()[1];
+
+            if param_value.strip_prefix('u').is_some() {
+                // parse the bits length of the exponent from the immediate value.
+                let bits_len = parse_element_param(op, 1)?;
+                n = bits_len.as_int() as usize;
+            } else {
+                // parse immediate value.
+                let imm = parse_element_param(op, 1)?;
```

Comment:
Would be cool if we had special cases for things like `exp.1`, `exp.2` etc. For example, `exp.2` would be `dup mul`, `exp.3` could be `dup dup mul mul`. Probably makes sense to do it all the way up to `exp.6` or `exp.7`.